lib/doorkeeper/oauth/error_response.rb

```diff
@@ -41,7 +41,7 @@ def status
 
       def redirectable?
         name != :invalid_redirect_uri && name != :invalid_client &&
-          !URIChecker.oob_uri?(@redirect_uri)
+          !@redirect_uri.nil? && !URIChecker.oob_uri?(@redirect_uri)
```

Comment:
`!URIChecker&.oob_uri?(@redirect_uri)`

Reply:
Why we afraid that `URIChecker` doesn't have this method and need use safe-navigation operator? :thinking:

Reply:
I think it's safe enough to call this method with `nil` redirect uri @jasl , because of:

https://github.com/doorkeeper-gem/doorkeeper/blob/59a1227a963b1034bb9df61ad377bb745c34f297/lib/doorkeeper/oauth/helpers/uri_checker.rb#L95-L97

Reply:
Oh I just comment that `!@redirect_uri.nil? && !URIChecker.oob_uri?(@redirect_uri)` can refactor to `!URIChecker&.oob_uri?(@redirect_uri)` which more simplify.

Reply:
Once again - I don't understand why we do a safe-call of `oob_uri?` for `URIChecker`? Safe-navigation operator purposes is to do a safe calls against `nil` objects (_Maybe_ monad for poor people ©), but here `URIChecker` isn't a `nil:NilClass` :thinking: 

Also  @cincospenguinos I prefer `@redirect_uri.present?` over `!@redirect_uri.nil?`

Reply:
>  it assumes that the error response is redirectable even though redirect_uri is nil

@cincopenguinos you are right, current doorkeeper's code base will `redirectable even though redirect_uri is nil`. But not only here, it happens with another case, too.

So, I am going to fix it at here by re-ordering validation of pre_auth: https://github.com/doorkeeper-gem/doorkeeper/pull/1277/files#r310095549

Reply:
Awesome. Thank you guys so much for the quick response time! I look forward to that change